### DISCUSSION TITLE
Enabling haptic rumble on pedals through SDL

### DISCUSSION THREAD

**Discussion by tbalch-tri** [gotzl/hid-fanatecff#73] [Q&A] [answered]
I've been digging into the driver and can see that there is code to enable haptic rumble on the throttle and brake pedals of the clubsport pedals v3. I am working with a custom steering wheel driver that is sending force feedback through SDL_HapticNewEffect and other SDL Haptic commands. I tried running SDL_HapticRumblePlay, but that just plays haptic feedback in the steering wheel. I was wondering if you could give any insight on how your driver catches SDL commands to set rumble on the pedals (which are different axes of the event joystick)?

**Comment by gotzl** [comment, marked as answer]
Hi,
currently, rumble of pedals is not wired up to FFB linux API, so it can not work with SDL Haptic commands.
However, when you plug-in the pedals via USB directly to the PC, you'll get a `rumble` sysfs file. This file you can use to control rumble of the pedals by writing to it, see `ClubSport Pedals V3` section in the readme.

**Reply by tbalch-tri** [reply]
Awesome, thank you so much for your answer, that helps a lot and has saved me a lot of time!

**Reply by tbalch-tri** [reply]
Do you know how pedal rumble with the pedals connected to the steering wheel base could be made to work?

**Reply by gotzl** [reply]
Hmm... I don't have pedals with this functionality, I can't test it. Just to confirm, you've tested with separated pedals and this worked?
Chances are that the same USB packets also work with combined pedals/wheel-base. Can you switch to the `pedal_rumble` branch and after compiling (and possibly unloading the module with `rmmod hid_fanatec`) load the module with `insmod hid-fanatec.ko pedal_rumble=1`. This should give you a `pedal_rumble` file in sysfs that works similar to the `rumble` file from having the separated pedals.

**Reply by tbalch-tri** [reply]
I have not tested with the separated pedals, but I do know that if you plug the pedals in separately, they appear as a separate joystick device in ubuntu from the steering wheel. I assumed that plugging in the pedals separately from the wheel is what you meant when in your initial answer when you referred to the rumble sysfs appearing (then using the values listed in the readme). I'm switching gears to something else for the moment, but I can try that out later this week if I get some time.
  In order to trigger the pedal rumble, I have to manually set the values in the pedal_rumble sysfs file?

Thanks for your help. Also, are you open to PRs from others to this driver? If I don't get to it myself, we may be able to dedicate some time to addressing this, especially if you think it's possible/reasonably straightforward.

**Reply by gotzl** [reply]
>  I assumed that plugging in the pedals separately from the wheel is what you meant when in your initial answer when you referred to the rumble sysfs appearing (then using the values listed in the readme)

yes, exactly

> In order to trigger the pedal rumble, I have to manually set the values in the pedal_rumble sysfs file?

yes, s.t. like `echo 0xffff00 > padle_rumble` should activate gas/brake rumble ...

> are you open to PRs from others to this driver?

Sure, gladly!
